src/main/java/com/exasol/adapter/document/queryplanning/RemoteTableQueryFactory.java

```diff
@@ -31,8 +32,9 @@ public RemoteTableQuery build(final SqlStatement selectStatement, final String s
         try {
             selectStatement.accept(visitor);
         } catch (final AdapterException exception) {
-            // Should no happen since no adapter exceptions are thrown in the visitor
-            throw new IllegalStateException("Unexpected AdapterException: " + exception.getMessage(), exception);
+            throw new IllegalStateException(ExaError.messageBuilder("E-VSD-42").message(
+                    "Unexpected AdapterException. This should no happen since no adapter exceptions are thrown in the visitor.")
```

Comment:
I don't like all those messages about "should no happen ". I would remove it from messages